src/encoded/static/components/cart/description.js

```diff
@@ -0,0 +1,328 @@
+    const isDescriptionEmpty = sanitizeDescription(editedDescription) === '';
+    const isDescriptionSaveButtonDisabled = !isDescriptionEdited || (cart.status === 'listed' && isDescriptionEmpty);
+    return {
+        // Label that should appear in the Save Description button
+        descriptionSaveButtonLabel,
+        // True if Save Description button should appear disabled
+        isDescriptionSaveButtonDisabled,
+        // True if the description has been changed from the saved version
+        isDescriptionEdited,
+    };
+};
+
+
+export const CartDescriptionEditorContent = ({
+    editedDescription,
+    savedDescription,
+    isDescriptionEdited,
+    onChangeDescription,
+}) => {
+    /**
+     * Called when the users wants to reset the text area to the saved description.
+     */
+    const onResetDescription = () => {
+        onChangeDescription(savedDescription);
+    };
+
+    return (
+        <div className="cart-description-editor">
+            <textarea
+                id="text-editing-area"
+                value={editedDescription}
+                maxLength={MAX_DESCRIPTION_LENGTH}
+                rows="12"
+                name="cart-description-text-area"
+                onChange={(e) => onChangeDescription(e.target.value)}
+                className="cart-description__editor"
+            />
+            <div className="cart-description-editor__tools">
+                <button type="button" disabled={!isDescriptionEdited} className="btn btn-sm btn-danger" onClick={onResetDescription}>
+                    Revert to saved description
+                </button>
+                <TextAreaCounter value={editedDescription} maxLength={MAX_DESCRIPTION_LENGTH} />
+            </div>
+        </div>
+    );
+};
+
+CartDescriptionEditorContent.propTypes = {
+    /** Holds current edited description */
+    editedDescription: PropTypes.string,
+    /** Description as currently saved in the cart object */
+    savedDescription: PropTypes.string,
+    /** True if user has edited the description */
+    isDescriptionEdited: PropTypes.bool.isRequired,
+    /** Called as the user edits the description */
+    onChangeDescription: PropTypes.func.isRequired,
+};
+
+CartDescriptionEditorContent.defaultProps = {
+    editedDescription: '',
+    savedDescription: '',
+};
+
+
+/**
+ * Displays the formatted cart description.
+ */
+export const CartDescriptionDisplay = ({ description }) => {
+    /** Description cleaned of any dangerous HTML/Javascript */
+    const [sanitizedDescription, setSanitizedDescription] = React.useState(description);
+
+    React.useEffect(() => {
+        // Must have a DOM to sanitize the description.
+        setSanitizedDescription(marked(sanitizeDescription(description || '')));
+    }, [description]);
+
+    if (sanitizedDescription) {
+        return (
+            <div className="cart-description">
+                <div dangerouslySetInnerHTML={{ __html: sanitizedDescription }} />
```

Comment:
Is there another way?

Reply:
I think it’s likely it was named `dangerouslySetInnerHTML` because you _could_ pass it executable code, not that it conjures executable code in markup that has been stripped of it.